### DISCUSSION TITLE
Correct way to dump the postgres database?

### DISCUSSION THREAD

**Discussion by vitobotta** [immich-app/immich#6083] [Q&A]
Hi, I am trying to test restoring my backups and when I restore the database I get the following errors:

```
ERROR:  type "earth" does not exist
LINE 1: ...ians($1))*sin(radians($2))),earth()*sin(radians($1)))::earth
                                                                  ^
QUERY:  SELECT cube(cube(cube(earth()*cos(radians($1))*cos(radians($2))),earth()*cos(radians($1))*sin(radians($2))),earth()*sin(radians($1)))::earth
CONTEXT:  SQL function "ll_to_earth" during inlining
COPY geodata_places, line 1: "3038999	Soldeu	1.66769	42.57688	AD	02		2017-11-06"
.....
ERROR:  type "vector" does not exist
```

What is the correct way to back up the database to ensure a working restore? At the moment I am dumping the db with this command:

```bash
pg_dump --clean -h 127.0.0.1 -p 5432 -U postgres $DATABASE > $DUMP_PATH
```

Thanks!

**Comment by danieldietzler** [comment]
Hey, please refer to https://immich.app/docs/administration/backup-and-restore for details on backing up the database :)

**Reply by vitobotta** [reply]
Hi, I am doing the same thing basically, just dumping a single database instead of all. I am testing with a new container with the same postgres image and without any volumes etc so it starts from scratch as that document suggests. What else can I try?

**Reply by danieldietzler** [reply]
Oh ok, sorry for drawing conclusions too quickly.
@mertalev May help with some SQL sorcery then :)

**Reply by vitobotta** [reply]
I also tried with different dumping formats (like directory etc) but same issue. Happy new year btw! :)

**Reply by danieldietzler** [reply]
Haha still two hours to go here :D
You too though!

**Reply by mertalev** [reply]
I believe this is related to issue #5630. [This](https://github.com/immich-app/immich/issues/5630#issuecomment-1866581570) comment might be what you're looking for

**Reply by vitobotta** [reply]
I was just coming to post to say  that I found this! Thanks!
